vnext/Microsoft.ReactNative/Modules/I18nManagerModule.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "I18nManagerModule.h"
+#include <IReactDispatcher.h>
+#include <winrt/Windows.ApplicationModel.Resources.Core.h>
+#include <winrt/Windows.Globalization.h>
+#include "Unicode.h"
+
+namespace Microsoft::ReactNative {
+
+static const React::ReactPropertyId<bool> &SystemIsRTLPropertyId() noexcept {
+  static const React::ReactPropertyId<bool> prop{L"I18n", L"SystemIsRTL"};
+  return prop;
+}
+
+static const React::ReactPropertyId<bool> &AllowRTLPropertyId() noexcept {
+  static const React::ReactPropertyId<bool> prop{L"I18n", L"AllowRTL"};
+  return prop;
+}
+
+static const React::ReactPropertyId<bool> &ForceRTLPropertyId() noexcept {
+  static const React::ReactPropertyId<bool> prop{L"I18n", L"ForceRTL"};
+  return prop;
+}
+
+void I18nManager::InitI18nInfo(const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto layoutDirection =
+      winrt::Windows::ApplicationModel::Resources::Core::ResourceContext().GetForCurrentView().QualifierValues().Lookup(
+          L"LayoutDirection");
+
+  propertyBag.Set(SystemIsRTLPropertyId(), layoutDirection != L"LTR");
+}
+
+/*static*/ bool I18nManager::IsRTL(const React::ReactPropertyBag &propertyBag) noexcept {
+  if (propertyBag.Get(ForceRTLPropertyId()))
+    return true;
+
+  if (!propertyBag.Get(AllowRTLPropertyId()))
+    return false;
+
+  return !!propertyBag.Get(SystemIsRTLPropertyId());
+}
+
+void I18nManager::AllowRTL(bool allowRTL) noexcept {
+  m_context.Properties().Set(AllowRTLPropertyId(), allowRTL);
+}
+
+void I18nManager::ForceRTL(bool forceRTL) noexcept {
+  m_context.Properties().Set(ForceRTLPropertyId(), forceRTL);
+}
+
+void I18nManager::SwapLeftAndRightInRTL(bool /*flipStyles*/) noexcept {
+  // TODO - https://github.com/microsoft/react-native-windows/issues/4662
+}
+
+void I18nManager::GetConstants(React::ReactConstantProvider &provider) noexcept {
+  std::string locale = "en-us";
+
+  auto langs = winrt::Windows::Globalization::ApplicationLanguages::Languages();
+  if (langs.Size() > 0) {
+    locale = Microsoft::Common::Unicode::Utf16ToUtf8(langs.GetAt(0));
+  }
+
+  provider.Add(L"localeIdentifier", locale);
+  provider.Add(L"doLeftAndRightSwapInRTL", false);
+  provider.Add(L"isRTL", IsRTL(m_context.Properties()));
```

Comment:
why is RTL-ness not a function of the locale?

Reply:
I suspect it is under the covers.  The description for the layoutdirection `ResourceContext.QualifierValues` is:

```
https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.resources.core.resourcecontext.qualifiervalues?view=winrt-18362
```
https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.resources.core.resourcecontext.qualifiervalues?view=winrt-18362

But I dont see a way to get it except from the UI thread -- which is a shame.